mne/stats/regression.py

```diff
@@ -392,3 +392,254 @@ def _make_evokeds(coefs, conds, cond_length, tmin_s, tmax_s, info):
+        X_names = []
+        if events is not None:
+            X_ev, X_ev_names = _events_to_continuous(raw, events, event_id)
+            X.append(X_ev)
+            X_names = X_names + X_ev_names
+        elif continuous is not None:
+            if continuous_names is None:
+                continuous_names = ['feat_%s' % ii
+                                    for ii in range(continuous.shape[0])]
+            if continuous.shape[-1] != raw.n_times:
+                raise ValueError('Continuous data must have same'
+                                 ' n_times as raw.')
+            if len(continuous_names) != continuous.shape[0]:
+                raise ValueError(
+                    'feat_names and X.shape[0] must be the same size')
+            X.append(continuous)
+            X_names = X_names + continuous_names
+        else:
+            raise ValueError('Must supply either events or continuous inputs')
+        X = np.vstack(X)
+
+        # Delay X
+        X_delayed = delay_timeseries(X, self.sfreq, self.delays)
+        X_names = [(iname, idelay)
+                   for iname in X_names for idelay in self.delays]
+        self.coef_names_ = np.array(X_names).T
+
+        # Fit the model and assign coefficients
+        X_delayed = X_delayed[:, fit_ixs]
+        Y = raw._data[:, fit_ixs]
+        self.ch_names = [raw.ch_names[ii] for ii in picks]
+        self.coef_ = np.zeros([len(picks), X_delayed.shape[0]])
+        self.X_ = X_delayed
+        for ii, ipick in enumerate(picks):
+            y = Y[ipick][:, np.newaxis]
+            if self.preproc_y:
+                y = self.est._pre_transform(y)[0]
+            self.est.fit(X_delayed.T, y)
+            self.coef_[ii, :] = self.est.steps[-1][-1].coef_
+
+    def predict(self, X):
+        """Generate predictions using fit coefficients.
+
+        This uses the `coef_` attribute for predictions.
+        """
+        X_lag = delay_timeseries(X, self.sfreq, self.delays)
+
+        Xt = self.est._pre_transform(X_lag.T)[0]
+        preds = np.dot(self.coef_, Xt.T)
+        return preds
+
+    def coefs_to_series(self):
+        """Return the raw coefficients as a pandas series.
+
+        Outputs
+        -------
+        sr : pandas Series, shape (n_coefficients,)
+            The coefficients as a pandas series object.
+        """
+        import pandas as pd
+        if not hasattr(self, 'coef_'):
+            raise ValueError('Fit the model first...')
+        ix = pd.MultiIndex.from_arrays(self.coef_names,
+                                       names=['feat', 'lag'])
+        sr = []
+        for ich, icoef in enumerate(self.coefs_):
+            isr = pd.DataFrame(icoef[:, np.newaxis], index=ix)
+            isr['ch'] = self.ch_names[ich]
+            isr = isr.set_index('ch', append=True).squeeze()
+            sr.append(isr)
+        sr = pd.concat(sr, axis=0)
+        return sr
+
+
+def delay_timeseries(ts, sfreq, delays):
+    """Return a time-lagged input timeseries.
```

Comment:
Also it seems very redundant with `mne.stats.regression._prepare_rerp_preds`. We should probably only have one or the other (depending on which one is faster).